src/rtctools_heat_network/heat_physics_mixin.py

```diff
@@ -2384,6 +2384,108 @@ class and the selected temperature in the network.
+
+            sec_heat = self.state(f"{hp}.Secondary_heat")
+            elec = self.state(f"{hp}.Power_elec")
+            nominal = self.variable_nominal(f"{hp}.Secondary_heat")
+
+            if (
+                len(sec_sup_temps) <= 1
+                and len(sec_ret_temps) <= 1
+                and len(prim_sup_temps) <= 1
+                and len(prim_ret_temps) <= 1
+            ):
+                cop = parameters[f"{hp}.COP"]
+                constraints.append(((sec_heat - cop * elec) / nominal, 0.0, 0.0))
+            else:
+                big_m = 2.0 * self.bounds()[f"{hp}.Secondary_heat"][1]
+                for sec_sup_temp in (
+                    sec_sup_temps
+                    if len(sec_sup_temps) > 0
+                    else [parameters[f"{hp}.Secondary.T_supply"]]
+                ):
+                    sec_sup_selected = (
+                        1.0 - self.state(f"{sec_sup_carrier}_{sec_sup_temp}")
+                        if len(sec_sup_temps) > 0
+                        else 0
+                    )
+                    for sec_ret_temp in (
+                        sec_ret_temps
+                        if len(sec_ret_temps) > 0
+                        else [parameters[f"{hp}.Secondary.T_return"]]
+                    ):
+                        sec_ret_selected = (
+                            1.0 - self.state(f"{sec_ret_carrier}_{sec_ret_temp}")
+                            if len(sec_ret_temps) > 0
+                            else 0
+                        )
+                        for prim_sup_temp in (
+                            prim_sup_temps
+                            if len(prim_sup_temps) > 0
+                            else [parameters[f"{hp}.Primary.T_supply"]]
+                        ):
+                            prim_sup_selected = (
+                                1.0 - self.state(f"{prim_sup_carrier}_{prim_sup_temp}")
+                                if len(prim_sup_temps) > 0
+                                else 0
+                            )
+                            for prim_ret_temp in (
+                                prim_ret_temps
+                                if len(prim_ret_temps) > 0
+                                else [parameters[f"{hp}.Primary.T_return"]]
+                            ):
+                                prim_ret_selected = (
+                                    1.0 - self.state(f"{prim_ret_carrier}_{prim_ret_temp}")
+                                    if len(prim_ret_temps) > 0
+                                    else 0
+                                )
+                                efficiency = parameters[f"{hp}.efficiency"]
+                                t_cond = 273.15 + sec_sup_temp
+                                t_evap = 273.15 + prim_sup_temp
+
+                                cop_carnot = efficiency * t_cond / (t_cond - t_evap)
+                                selected = (
```

Comment:
add comment that selected is 0 when all the tempeartures are selected, just based on the names and not looking back how you configered them one would actually expect the opposite

Reply:
changed the name to not_selected